custom_components/ttlock/api.py

```diff
@@ -142,11 +143,25 @@ def lock_connectable(lock) -> bool:
             )
             return has_gateway or has_wifi
 
-        return [lock["lockId"] for lock in res["list"] if lock_connectable(lock)]
+        locks = [lock["lockId"] for lock in res_local["list"] if lock_connectable(lock)]
+        locks += [lock["lockId"] for lock in res_remote["list"] if lock_connectable(lock)]
+
+        return locks
 
     async def get_lock(self, lock_id: int) -> Lock:
         """Get a lock by ID."""
-        res = await self.get("lock/detail", lockId=lock_id)
+        try:
+            res = await self.get("lock/detail", lockId=lock_id)
+            return Lock.parse_obj(res)
+        except RequestFailed:
+            # It is either eKey lock or really exception. For eKey locks this
+            # API will return errcode if you are not admin for this lock.
+            # Give up here and try another API
+            pass
+
+        # Don't wrap it in try-except, in case of exception it will reraise the
+        # same one you got in the first request
+        res = await self.get("key/get", lockId=lock_id)
```

Comment:
For users with a lot of locks there are already some problems with hitting the rate limits of the API.

Is there a way we could potentially track what kind of lock we're referencing so that features can make the right decision rather than having to try both APIs?

One idea might be to do something like:

- Update `get_locks` to return instances of a class instead of ints like `LockId(id=123, type=LockType.local)`
- `get_lock` can call the correct api directly
- Add a `type` argument to the `Lock` class somehow too?